### PR TITLE
Configurable timeout

### PR DESCRIPTION
**Pull request by nvthongswansea**
This PR will allow users to set a general timeout for gridscale terraform provider via tf variable `timeout` (or `GRIDSCALE_TIMEOUT_SECONDS` env var). As a result, Terraform will use a custom gsclient-go client instead of the default one. Furthermore, there are some small changes (which don't affect much) such as `api_url` will be optional.

### CODE REVIEW

**Comment by itakouna** [comment]
@nvthongswansea  Thanks for the PR. `https://www.terraform.io/docs/configuration/resources.html#operation-timeouts` this timeouts feature is specific for AWS resource. That why I've asked to implement resource specific timeout.

File: gridscale/provider.go [hunk 1]
```diff
@@ -22,10 +22,16 @@ func Provider() terraform.ResourceProvider {
 			},
 			"api_url": {
 				Type:        schema.TypeString,
-				Required:    true,
-				DefaultFunc: schema.EnvDefaultFunc("GRIDSCALE_URL", "https://api.gridscale.io"),
+				Optional:    true,
+				DefaultFunc: schema.EnvDefaultFunc("GRIDSCALE_URL", nil),
 				Description: "the url for the gridscale API.",
 			},
+			"gsc_timeout_secs": {
```
Comment (itakouna) [inline gridscale/provider.go]: I would simply name it `timeout` and in docs. we should show the time unit in seconds

**Comment by nvthongswansea** [comment]
> @nvthongswansea Thanks for the PR. `https://www.terraform.io/docs/configuration/resources.html#operation-timeouts` this timeouts feature is specific for AWS resource. That why I've asked to implement resource specific timeout.

Actually it applies for all tf providers that declare `Timeout` blocks in their resources. I already removed that block before, I will bring it back. E.g https://github.com/terraform-providers/terraform-provider-aws/blob/620769e2d2541450f22d739029cdc2f3de376609/aws/resource_aws_db_instance.go#L38

**Comment by nvthongswansea** [comment]
@itakouna I have checked again. The current version of gsclient-go only allows us to set the timeout for the whole client, and tf does not use the `timeout` variables (they gives us that `Timeout` block but we have to implement it manually by pulling timeout vars, e.g `d.Timeout(schema.TimeoutCreate)` , and passing these values to gsclient-go functions - which is not possible ATM). So I will remove all comments about specific timeouts, but the global one.

**Review by itakouna — APPROVED** [review]
LGTM

**Comment by bkircher** [comment]
Ditto :+1:

**Comment by bkircher** [comment]
Thanks @nvthongswansea !

**Pull request merged**